vcbuild.bat

```diff
@@ -292,10 +292,19 @@ goto exit
 @rem Visual Studio v17.10 has a bug that causes the build to fail.
 @rem Check if the version is v17.10 and exit if it is.
 echo %VSCMD_VER% | findstr /b /c:"17.10" >nul
-if %errorlevel% neq 1  (
+if %errorlevel% neq 1 (
   echo Node.js doesn't compile with Visual Studio 17.10 Please use a different version.
   goto exit
 )
+@rem Same applies to v17.12 for MSVC.
+echo %VSCMD_VER% | findstr /b /c:"17.12" >nul
+if %errorlevel% neq 1 (
+  @rem Clang 18.1.8 Provided with VS 17.12 works fine.
+  if not defined clang_cl (
+    echo Node.js doesn't compile with Visual Studio 17.12 Please use a different version.
```

Comment:
Including a link to an issue that describes the reason for this would be helpful.

Reply:
Hey @jasnell thanks for the suggestion. That makes a lot of sense. However, you've already approved this, the CI has finished, and based on the [MSFT issue](https://developercommunity.visualstudio.com/t/Regression-in-MSVC-v1712:-Compilation-E/10794016) the reason behind it is fixed and pending release. I assume this will be in some 17.12.x release, so I'll most likely end up reverting this then. With all that in mind, would you oppose me landing this PR as is?